### ISSUE TITLE
Develop Branch Policy Update Breaks Ford PTSnapshot Process

### ISSUE THREAD

**Issue by JackLivio** [smartdevicelink/sdl_core#402]
I am seeing some differences for the policy table update process in the Develop and Master Branch.

[Commit to develop branch in question] (https://github.com/smartdevicelink/sdl_core/commit/5abb7cdca5e23e9765b616fef5440fe4b0e49d31)

Changes to ../mobile/on_system_request_notification.cc: A part of code was removed where core reads the file contents of the filename passed as a parameter from the hmi. Reading in this file's contents is essential to the Ford Process of sending an encrypted snapshot.

I see the comments included in this commit that say
```
 +    /* According to requirements:
 +       "If the requestType = PROPRIETARY, add to mobile API fileType = JSON
 +        If the requestType = HTTP, add to mobile API fileType = BINARY"
 +       Also in Genivi SDL we don't save the PT to file - we put it directly in binary_data */
```

Highlighting the last comment, is this repository only supposed to follow Genivi Requirements? What about supporting existing Ford / legacy processes?

**Comment by JackLivio** [comment]
Here is a diagram of the ford process that this commit breaks.
<img width="1198" alt="screen shot 2016-03-02 at 5 00 58 pm" src="https://cloud.githubusercontent.com/assets/12038416/13477203/3c9192f2-e099-11e5-90a2-c6e9388a3c99.png">

**Comment by madhuy** [comment]
with develop branch PT , even it is not working non-ford version PT update . because mobile proxies expecting FORD specific format in JSON (ex: HTTPRequest, headers,body). 
https://github.com/smartdevicelink/sdl_ios/blob/develop/SmartDeviceLink-iOS/SmartDeviceLink/SDLProxy.m#L361

https://github.com/smartdevicelink/sdl_ios/blob/develop/SmartDeviceLink-iOS/SmartDeviceLink/SDLProxy.m#L476

**Comment by JackLivio** [comment]
@madhuy: We are working on adding HTTP SystemRequest (Genivi) support to the ios proxy today. We plan to have the mobile libraries construct the HTTP Request header and body for http onSystemRequests.

**Comment by madhuy** [comment]
Thanks, I modified SDLproxy locally it working for me . 

And also I noticed , SDL core master branch does not work ford specific implementation also. 
Ford specific PT exchange start from PTExchangeHandlerExt::Start() (pt_exchange_handler_ext.cc). this file not been added in make for compiling.  

Please check this as well.

**Comment by theresalech** [comment]
Fixed with [Release 4.0.12](https://github.com/smartdevicelink/sdl_core/releases/tag/4.0.12)
